klp_tc_functions.sh

```diff
@@ -378,5 +379,10 @@ if [ ! -f $KLP_ENV_CACHE_FILE ]; then
     # based on presence of an obsolete API function
     echo -n 'export KLP_TEST_USE_OLD_REG_API=' >> $KLP_ENV_CACHE_FILE
     ( grep -q 'T klp_register_patch$' /proc/kallsyms && echo "1" || echo "0" ) >> $KLP_ENV_CACHE_FILE
+
+    # Decide whether to use hrtimer_setup_sleeper_on_stack() API
+    # (kernel commit c9bd83abfeb9)
+    echo -n 'export KLP_TEST_USE_NEW_HRTIMER_SETUP_API=' >> $KLP_ENV_CACHE_FILE
+    ( grep -q 'T hrtimer_setup_sleeper_on_stack$' /proc/kallsyms && echo "1" || echo "0" ) >> $KLP_ENV_CACHE_FILE
```

Comment:
OLD_HRTIMER_API uses a compile test. I do not really have a preference but it would be nice to be consistent.

Reply:
I did try using compile time test. But, thought this approach is a  bit smaller.

Reply:
It is. So if you could remove the compile test for OLD_HRTIMER_API and replace it with something like this (git grep) in a separate commit, it would be really great.

Reply:
Looks like compile time test was added on purpose. The number of arguments passed to hrtimer_init_sleeper() was changed from 2 to 1.  
Linux reference commit : b74494872555 ("hrtimer: Remove task argument from hrtimer_init_sleeper()")
Old API used 2 arguments in hrtimer_init_sleeper().

I think, compile time test seems to check if it generates compilation error when two arguments are passed in hrtimer_init_sleeper(). Then, better to not remove the compile time check for OLD_HRTIMER_API.

Reply:
Ok, thanks for checking. So let's go with your original proposal.